include/aws/s3/s3_client.h

```diff
@@ -150,6 +149,22 @@ enum aws_s3_checksum_algorithm {
     AWS_SCA_COUNT,
 };
 
+/* Keepalive properties are TCP only.
+ * If interval or timeout are zero, then default values are used.
+ */
+struct aws_s3_tcp_keep_alive_options {
```

Comment:
why break these out into their own struct?

But I COULD see an argument for a new struct for ALL connection-specific options, like 
`struct aws_s3_client_connection_options`. The argument being: if we want to allow endpoint-specific options in the future, now we can just re-use this struct that we were already using.

I'm wishy washy on whether this is a good idea. We'd have to move 2 existing variables into this new struct: `aws_s3_client_config.tls_mode` and `aws_s3_client_config.tls_connection_options`

Reply:
I have updated it to a pointer, and null means keep alive is false. I broke these out into their own struct because of their relationship to keep alive.